tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java

```diff
@@ -0,0 +1,377 @@
+
+        // invoke a remote service which runs in the callers context, ie do set the context header
+        remote = restPutInvocation(active, TCK_NON_PARTICIPANT_RESOURCE_PATH, SUPPORTS_PATH, "");
+        assertEquals("contextCheck4: the remote service should have ran with the active context", remote, active);
+        assertEquals("contextCheck4: after calling a remote service the active LRAs is different", active, getActiveLRA());
+
+        return Response.ok().entity(lraId).build();
+    }
+
+    /**
+     * Query the current value of a metric in the context of an LRA
+     * @param lraId the LRA that owns the metric
+     * @param metric the name of the metric
+     * @return the current value of the metric
+     */
+    @GET
+    @Path(METRIC_PATH + "/{metric}")
+    public Response count(@HeaderParam(LRA_TCK_HTTP_CONTEXT_HEADER) String lraId,
+                          @PathParam("metric") String metric) {
+        if (metrics.containsKey(lraId)) {
+            return Response.ok(metrics.get(lraId).get(metric)).build();
+        }else {
+            return Response.ok(-1).build();
+        }
+    }
+
+    @Leave
+    @PUT
+    @Path(LEAVE_PATH)
+    public Response leave(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId) {
+        return Response.ok().entity(lraId).build();
+    }
+
+    @PUT
+    @Path("/compensate")
+    @Compensate
+    public Response compensateWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId)
+            throws NotFoundException {
+        incrementMetric(Compensate.class.getName(), lraId);
+
+        return getEndPhaseResponse(false);
+    }
+
+    @PUT
+    @Path("/complete")
+    @Complete
+    public Response completeWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId)
+            throws NotFoundException {
+        incrementMetric(Complete.class.getName(), lraId);
+
+        return getEndPhaseResponse(true);
+    }
+
+    @Status
+    @GET
+    @Path(STATUS_PATH)
+    public Response status(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId) {
+        incrementMetric(Status.class.getName(), lraId);
+
+        return Response.status(endPhaseStatus).entity(status.name()).build();
+    }
+
+    @Forget
+    @DELETE
+    @Path("/forget")
+    public Response forget(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId) {
+        incrementMetric(Forget.class.getName(), lraId);
+
+        return Response.status(endPhaseStatus).entity(status.name()).build();
+    }
+
+    // clear any injected participant behaviour
+    @PUT
+    @Path(STATUS_PATH)
+    public Response clearStatus(@HeaderParam(LRA_TCK_HTTP_CONTEXT_HEADER) String lraId,
+                           @HeaderParam(LRA_TCK_FAULT_TYPE_HEADER) String tckFaultType) {
+        switch (status) {
+            case Compensating:
+                status = ParticipantStatus.Compensated;
+                break;
+            case Completing:
+                status = ParticipantStatus.Completed;
+                break;
+            default:
```

Comment:
When no default, we should not specify this.

Reply:
I was keeping my IDE sweet but okay I can remove it.

Reply:
@rdebusscher I added the checkstyle rule `WhitespaceAround` to check for spaces around else: 
```
                                <module name="WhitespaceAround">
                                  <property name="tokens" value="LITERAL_ELSE"/>
                                </module>
```
but the rule is also checks for the presence of the `default` branch on `switch` statements so I kept the default branch.